core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityExpressionValidatorSpec.scala

```diff
@@ -0,0 +1,193 @@
+
+class DataRepoEntityExpressionValidatorSpec extends FlatSpec with TestDriverComponent with RawlsTestUtils with ScalaFutures with DataRepoEntityProviderSpecSupport with DataRepoExpressionFixture {
+
+  def toExpressionMap(expressions: Seq[String]): Map[String, AttributeString] =
+    expressions.map { expr => expr.toString -> AttributeString(expr) }.toMap
+
+  def toMethodInput(tuple: (String, AttributeString)): MethodInput = tuple match {
+    case (name, expr) => {
+      val valueType = new ValueType()
+      valueType.setTypeName(TypeNameEnum.STRING)
+      val input = new ToolInputParameter()
+      input.setName(name)
+      input.setValueType(valueType)
+      MethodInput(input, expr.value)
+    }
+  }
+
+  def toMethodInputs(methodConfiguration: MethodConfiguration) : Seq[MethodInput] = {
+    methodConfiguration.inputs.map(toMethodInput).toSeq
+  }
+
+  def toGatherInputs(exprs: Map[String, AttributeString]): GatherInputsResult = {
+    val methodInputs = exprs.map(toMethodInput)
+    GatherInputsResult(methodInputs.toSet, Set(), Set(), Set())
+  }
+
+  val rootTableName = "rootTable"
+  // todo: how to keep this in sync with the fixtures...
+  val rootTableColumns = List("gvcf", "with-dash", "library:cohort", "library:cohort1", "arbitrary:whatever", "underscores_are_ok", "_", "case_sample")
+  val tables: List[TableModel] = List(
+    new TableModel().name(rootTableName).columns(rootTableColumns.map(new ColumnModel().name(_)).asJava)
+  )
+  val provider = createTestProvider(dataRepoDAO = new SpecDataRepoDAO(Right(createSnapshotModel(tables))))
+  val expressionValidator: ExpressionValidator = provider.expressionValidator
+
+  val allValid = MethodConfiguration("dsde", "methodConfigValidExprs", Some(rootTableName), prerequisites=None,
+    inputs = toExpressionMap(validInputExpressions),
+    outputs = toExpressionMap(validOutputExpressions),
+    AgoraMethod("dsde", "three_step", 1))
+
+  val allValidNoRootMC = allValid.copy(inputs = toExpressionMap(validInputExpressionsWithNoRoot), outputs = toExpressionMap(validWorkspaceOutputExpressions), rootEntityType = None)
+
+  val allInvalid = MethodConfiguration("dsde", "methodConfigInvalidExprs", Some(rootTableName), prerequisites=None,
+    inputs = toExpressionMap(badInputExpressionsWithRoot),
+    outputs = toExpressionMap(invalidOutputExpressions),
+    AgoraMethod("dsde", "three_step", 1))
+
+  val allInvalidNoRootMC = allInvalid.copy(inputs = toExpressionMap(badInputExpressionsWithNoRoot), outputs = toExpressionMap(invalidOutputExpressions), rootEntityType = None)
+
+  val emptyExpr = "this.empty" -> AttributeString("")
+
+  val oneEmpty = MethodConfiguration("dsde", "methodConfigEmptyExpr", Some(rootTableName), prerequisites=None,
+    inputs = toExpressionMap(validInputExpressions) + emptyExpr,
+    outputs = toExpressionMap(validOutputExpressions),
+    AgoraMethod("dsde", "three_step", 1))
+
+
+  it should "validateAndParseMCExpressions" in {
```

Comment:
Some nits here seeing as we're creating this file now.  I know it was copied from the existing validator spec, but easier to clean this up now than later (if we want it cleaned up at all):

- rename `validateAndParseMCExpressions` - if anything, I would say we're just validating and the parsing is just a necessary component of validating.  Anyway, naming of tests would become more expressive if we do the following too:
- Can we break each assertion block below into its own test?  It seems unnecessary to have them all in one test.  If an assertion fails, processing of remaining tests won't happen.  Maybe that's what we want, but maybe not.  

Fully acknowledging these are nits and if you don't want to worry about them now, that's fine.

Reply:
Good point, I will rename `validateAndParseMCExpressions` in the test. It's named after the function in `ExpressionValidator` which we have since changed to `validateMCExpressions`, so no point in keeping around the old and inaccurate function name.

As for breaking up the assertions, I think that's a good idea. It'll let us better describe what each assertion is meant to test and that'll make test failures more descriptive as well.